R/select_nse.R

```diff
@@ -75,6 +75,22 @@
     out <- setdiff(out, grp_vars)
   }
 
+  # for named character vectors, we offer the service to rename the columns
+  if (typeof(expr_select) == "language") {
```

Comment:
I think we should have an additional arg in `.select_nse()` like `allow_rename = FALSE` by default and we set it to `TRUE` when we call `data_select()`. Then we can have `allow_rename && typeof(expr_select) == "language"` to avoid evaluating this branch for all the other contexts in which `.select_nse()` is called.